### PR TITLE
remove pipeline runs after once the build is finished

### PR DESCRIPTION
**Pull request by hjmodi**
* CLOUDBLD-10154

Signed-off-by: Harsh Modi <hmodi@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: tests/test_tekton.py [hunk 1]
```diff
@@ -411,6 +411,18 @@ def test_start_pipeline(self, openshift,
...
+    @responses.activate
+    def test_remove_pipeline(self, openshift):
+        p_run = PipelineRun(os=openshift, pipeline_run_name=PIPELINE_RUN_NAME)
+        responses.add(
+            responses.DELETE,
+            f'https://openshift.testing/apis/tekton.dev/v1beta1/namespaces/'
+            f'{TEST_OCP_NAMESPACE}/pipelineruns/{PIPELINE_RUN_NAME}',
+            json={}
+        )
+        p_run.remove_pipeline_run()
```
Comment (tkdchen) [inline tests/test_tekton.py:423]: Return value is not tested here.

**Review by tkdchen — APPROVED** [review]
:+1: Just leave one minor comment.

File: osbs/conf.py [hunk 2]
```diff
@@ -227,6 +227,10 @@ def get_kerberos_principal(self):
     def get_kerberos_ccache(self):
         return self._get_value("kerberos_ccache", self.conf_section, "kerberos_ccache")
 
+    def get_cleanup(self):
+        return self._get_value("cleanup", self.conf_section, "cleanup",
```
Comment (chmeliik) [inline osbs/conf.py:231]: I'd suggest to use a less vague name and/or document the option in https://osbs.readthedocs.io/en/master/admins.html#configuring-osbs-client

**Comment by hjmodi** [comment]
https://github.com/containerbuildsystem/osbs-docs/pull/216 created

File: osbs/conf.py (hunk 2 above)
Comment (hjmodi) [reply osbs/conf.py]: updated name and created docs PR

File: tests/test_tekton.py (hunk 1 above)
Comment (chmeliik) [reply tests/test_tekton.py:430]: I think cqi meant that this unit test should check that `remove_pipeline_run()` returns the JSON response

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Comment by chmeliik** [comment]
Coverage dropped but it's just the CLI, which we don't seem to test at all

File: tests/test_tekton.py (hunk 1 above)
Comment (hjmodi) [reply tests/test_tekton.py]: extended test to check the response

**Pull request merged**